src/devices/sunricher.ts

```diff
@@ -56,6 +68,20 @@ async function syncTime(endpoint: Zh.Endpoint) {
 }
 
 const definitions: Definition[] = [
+    {
+        zigbeeModel: ['HK-SENSOR-4IN1-A'],
+        model: 'HK-SENSOR-4IN1-A',
+        vendor: 'Sunricher',
+        description: '4IN1 Sensor',
+        extend: [
+            battery(),
+            identify(),
+            occupancy(),
+            iasZoneAlarm({'zoneType': 'generic', 'zoneAttributes': ['alarm_1', 'alarm_2', 'tamper', 'battery_low']}),
```

Comment:
```suggestion
            iasZoneAlarm({'zoneType': 'occupancy', 'zoneAttributes': ['alarm_1', 'tamper', 'battery_low']}),
```

Missed this initially, but can you try with this and remove `occupancy()`?

Reply:
Hello, thanks for your help. After removing the `occupancy()`, I took several tests and found that the `iasZoneAlarm()`  on this device isn't working properly, the device won't report occupancy status. 
I retained the `occupancy()` and removed the `iasZoneAlarm()`, and now the device is working properly.